provider/scaleway/scaleway_discover.go

```diff
@@ -0,0 +1,83 @@
+// Package scaleway provides node discovery for Scaleway.
+package scaleway
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+
+	"github.com/nicolai86/scaleway-sdk/api"
+)
+
+type Provider struct{}
+
+func (p *Provider) Help() string {
+	return `Scaleway:
+
+    provider:     "scaleway"
+    organization: The Scaleway organization access key
+    tag_name:     The tag name to filter on
+    token:        The Scaleway api access token
+    region:       The Scalway region
+`
+}
+
+func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error) {
+	if args["provider"] != "scaleway" {
+		return nil, fmt.Errorf("discover-scaleway: invalid provider " + args["provider"])
+	}
+
+	if l == nil {
+		l = log.New(ioutil.Discard, "", 0)
+	}
+
+	organization := args["organization"]
+	tagName := args["tag_name"]
+	token := args["token"]
+	region := args["region"]
+
+	l.Printf("[INFO] discover-scaleway: Organization is %q", organization)
+	l.Printf("[INFO] discover-scaleway: Region is %q", region)
+
+	// Create a new API client
+	api, err := api.New(
+		organization,
+		token,
+		region,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("discover-scaleway: %s", err)
+	}
+
+	// Currently fetching all servers since the API doesn't support
+	// filter options
+	servers, err := api.GetServers(true, 0)
+	if err != nil {
+		return nil, fmt.Errorf("discover-scaleway: %s", err)
+	}
+
+	// Get a list of private ips that match the tag name
+	return filterServersForTagName(servers, tagName, l), nil
```

Comment:
add DEBUG log line like https://github.com/hashicorp/go-discover/blob/master/provider/aws/aws_discover.go#L161

Reply:
Added at the end of `filterServersForTagName` https://github.com/anubhavmishra/go-discover/blob/e742c50534c126f9455e1f5b8e8a96274d17f610/provider/scaleway/scaleway_discover.go#L74